Dockerfile

```diff
@@ -11,30 +11,26 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-#
-
-ARG GOPROXY="https://goproxy.cn"
 
+ARG GOPROXY="https://goproxy.io,direct"
 # Build the openfunction binary
-FROM golang:1.16 as builder
-ARG GOPROXY
+FROM golang:1.17 as builder
+# Specify the image maintainer
+MAINTAINER openfuction.group
 
+# Specify the workspace
 WORKDIR /workspace
-# Copy the Go Modules manifests
-COPY go.mod go.mod
-COPY go.sum go.sum
+# Add the project packages to workspace
+ADD . /workspace
+
 # cache deps before building and copying source so that we don't need to re-download as much
 # and so that source changes don't invalidate our downloaded layer
-RUN go env -w GOPROXY=${GOPROXY} && go mod download
-
-# Copy the go source
-COPY main.go main.go
-COPY apis/ apis/
-COPY controllers/ controllers/
-COPY pkg/ pkg/
+# Refer to https://go.dev/ref/mod
+RUN go env -w GOPROXY=https://goproxy.io,direct && \
+    go mod tidy
 
-# Build
-RUN GOPROXY=${GOPROXY} CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -a -o openfunction main.go
+# Build openfunction binary.
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -ldflags '-s -w' -installsuffix cgo -o openfunction
```

Comment:
https://medium.com/@daviddavid533/cgo-enabled-0-and-installsuffix-cgo-are-no-longer-required-since-go-1-10-127d8c265cb1 according to this, `-installsuffix cgo` is no longer needed for go 1.10+